airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt

```diff
@@ -187,6 +192,50 @@ abstract class DestinationAcceptanceTest(
         streamSchema: JsonNode
     ): List<JsonNode>
 
+    private fun pruneAndMaybeFlatten(node: JsonNode): JsonNode {
+        val metaKeys =
+            mutableSetOf(
+                // V1
+                JavaBaseConstants.COLUMN_NAME_AB_ID,
+                JavaBaseConstants.COLUMN_NAME_EMITTED_AT,
+                // V2
+                JavaBaseConstants.COLUMN_NAME_AB_RAW_ID,
+                JavaBaseConstants.COLUMN_NAME_AB_EXTRACTED_AT,
+                JavaBaseConstants.COLUMN_NAME_AB_LOADED_AT,
+                JavaBaseConstants.COLUMN_NAME_AB_META,
+                JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID
+            )
+
+        val jsons = MoreMappers.initMapper().createObjectNode()
+        // Iterate over every key value pair in the json node
```

Comment:
checking my understanding: given this `node`:
```json
{
  "_airbyte_raw_id": "foo",
  "_airbyte_data": {
    "id": 42
  }
}
```

we would return this `jsons`:
```json
{
  "id": 42,
  "_airbyte_data": {
    "id": 42
  }
}
```

(assuming yes: are we intentionally promoting all the data fields, and also keeping the airbyte_data sub-object?)

... also, why not just this? is the idea to preserve non-airbyte top-level fields?
```kotlin
val newObjectNode = node["_airbyte_data"].deepCopy()
newObject.put("_airbyte_data", node["_airbyte_data"])
```

Reply:
You're correct, but that's a mistake. I'm missing an `else` near the end.

The intention is to remove airbyte fields AND promote anything under `_airbyte_data` to the top. I was trying to replicate the effective behavior of all the client implementations of the former `retrieveRecords` method.

Your example should produce.
```
{
  "id": 42
}
```